adapters/startio/startio.go

```diff
@@ -0,0 +1,179 @@
+	uri, err := url.ParseRequestURI(config.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	bidder := &StartioAdapter{
+		endpoint: uri.String(),
+	}
+
+	return bidder, nil
+}
+
+func (adapter *StartioAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var requests []*adapters.RequestData
+	var errors []error
+	requestCopy := *request
+
+	if err := validateRequest(requestCopy); err != nil {
+		return nil, []error{err}
+	}
+
+	validImpressions, err := getValidImpressions(requestCopy.Imp)
+	if err != nil {
+		return nil, []error{err}
+	}
+	for i := range validImpressions {
+		requestCopy.Imp = []openrtb2.Imp{validImpressions[i]}
+
+		requestBody, err := jsonutil.Marshal(requestCopy)
+		if err != nil {
+			errors = append(errors, fmt.Errorf("imp[%d]: failed to marshal request: %w", i, err))
+			continue
+		}
+
+		requestData := &adapters.RequestData{
+			Method:  http.MethodPost,
+			Uri:     adapter.endpoint,
+			Body:    requestBody,
+			Headers: buildRequestHeaders(),
+			ImpIDs:  []string{validImpressions[i].ID},
+		}
+
+		requests = append(requests, requestData)
+	}
+
+	return requests, errors
+}
+
+func (adapter *StartioAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	} else if response.StatusCode != http.StatusOK {
+		return nil, []error{wrapReqError(fmt.Sprintf("Unexpected status code: %d.", response.StatusCode))}
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
+		return nil, []error{err}
+	}
+
+	var bidResponse openrtb2.BidResponse
+	if err := jsonutil.Unmarshal(response.Body, &bidResponse); err != nil {
+		return nil, []error{wrapReqError(fmt.Sprintf("failed to unmarshal response body: %v", err))}
+	}
+
+	var errs []error
+	bidderResponse := adapters.NewBidderResponseWithBidsCapacity(len(bidResponse.SeatBid))
+
+	for i := range bidResponse.SeatBid {
+		for j := range bidResponse.SeatBid[i].Bid {
+			bid := &bidResponse.SeatBid[i].Bid[j]
+			bidType, err := getMediaTypeForBid(*bid)
+
+			if err != nil {
+				errs = append(errs, err)
+			} else {
+				bidderResponse.Bids = append(bidderResponse.Bids, &adapters.TypedBid{
+					Bid:     bid,
+					BidType: bidType,
+				})
+			}
+		}
+	}
+
+	return bidderResponse, errs
+}
+
+func validateRequest(request openrtb2.BidRequest) error {
+	if !isSupportedCurrency(request.Cur) {
+		return wrapReqError("unsupported currency: only USD is accepted")
+	}
+
+	if !hasSiteOrAppID(request) {
+		return wrapReqError("request must contain either site.id or app.id")
+	}
+
+	return nil
+}
+
+func getValidImpressions(imps []openrtb2.Imp) ([]openrtb2.Imp, error) {
+	var validImpressions []openrtb2.Imp
+
+	for _, imp := range imps {
+		imp.Audio = nil
```

Comment:
You can delete this line. There's no need to explicitly delete audio. PBS core will strip out any media type objects that your bidder has not declared support for in your YAML file.